### PR TITLE
[NAN-192] datadog add metric monitor for action successfailure

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Track action successes and failures

## Issue ticket number and link
NAN=192

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/server/lib/controllers/sync.controller.ts [hunk 1]
```diff
@@ -392,20 +393,33 @@ class SyncController {
...
-                res.send(actionResponse);
+                errorManager.errResFromNangoErr(res, actionResponse.err);
                 return;
             }
-        } catch (e) {
+        } catch (e: any) {
+            const content = e.message || 'Unknown error';
+
+            await metricsManager.capture(
+                MetricTypes.ACTION_SUCCESS,
```
Comment (bodinsamuel) [inline packages/server/lib/controllers/sync.controller.ts]:
```suggestion
                MetricTypes.ACTION_FAILURE,
```

File: packages/shared/lib/services/notification/slack.service.ts [hunk 2]
```diff
@@ -268,20 +273,20 @@ class SlackService {
...
             payload,
             activityLogId as number,
             environment_id
-        )) as ServiceResponse<SlackActionResponse>;
+        );
 
-        if (actionSuccess && actionResponse?.ts) {
-            await this.updateNotificationWithTimestamp(slackNotificationStatus.id, actionResponse?.ts as string);
+        if (isOk(actionResponse) && actionResponse.res.ts) {
+            await this.updateNotificationWithTimestamp(slackNotificationStatus.id, actionResponse.res.ts as string);
```
Comment (bodinsamuel) [inline packages/shared/lib/services/notification/slack.service.ts]:
```suggestion
            await this.updateNotificationWithTimestamp(slackNotificationStatus.id, actionResponse.res.ts);
```

File: packages/shared/lib/services/notification/slack.service.ts [hunk 3]
```diff
@@ -292,20 +297,20 @@ class SlackService {
             slackNotificationStatus.admin_slack_timestamp
         );
 
-        const content = actionSuccess
+        const content = isOk(actionResponse)
             ? `The action ${this.actionName} was successfully triggered for the ${flowType} ${syncName} for environment ${slackConnection?.environment_id} for account ${accountUUID}.`
-            : `The action ${this.actionName} failed to trigger for the ${flowType} ${syncName} with the error: ${actionError?.message} for environment ${slackConnection?.environment_id} for account ${accountUUID}.`;
+            : `The action ${this.actionName} failed to trigger for the ${flowType} ${syncName} with the error: ${actionResponse.err?.message} for environment ${slackConnection?.environment_id} for account ${accountUUID}.`;
```
Comment (bodinsamuel) [inline packages/shared/lib/services/notification/slack.service.ts]:
```suggestion
            : `The action ${this.actionName} failed to trigger for the ${flowType} ${syncName} with the error: ${actionResponse.err.message} for environment ${slackConnection?.environment_id} for account ${accountUUID}.`;
```

**Review by bodinsamuel — APPROVED** [review]
I can't test Datadog itself since I don't have the creds but the rest is working 👌🏻

**Review by TBonnin — COMMENTED** [review]
I would be tempted to always use traces rather than metrics. You have the advantage of getting hits/error/duration metrics for "free" while also having individual spans with dimensions that could be helpful for debugging purpose
Up to you though

**Comment by khaliqgant** [comment]
> I would be tempted to always use traces rather than metrics. You have the advantage of getting hits/error/duration metrics for "free" while also having individual spans with dimensions that could be helpful for debugging purpose Up to you though

Yeah, agree that is the next iteration of this. I'll either make a follow up or just add that in tomorrow. Thanks

File: packages/server/lib/controllers/sync.controller.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/server/lib/controllers/sync.controller.ts]: Whoops! Thanks. Updated with https://github.com/NangoHQ/nango/pull/1642/commits/7aa24a6b25495d62e32b6a234f6dfe6211103e84

File: packages/shared/lib/services/notification/slack.service.ts (hunk 2 above)
Comment (khaliqgant) [reply packages/shared/lib/services/notification/slack.service.ts]: Updated with https://github.com/NangoHQ/nango/pull/1642/commits/7aa24a6b25495d62e32b6a234f6dfe6211103e84

File: packages/shared/lib/services/notification/slack.service.ts (hunk 3 above)
Comment (khaliqgant) [reply packages/shared/lib/services/notification/slack.service.ts]: Updated with https://github.com/NangoHQ/nango/pull/1642/commits/7aa24a6b25495d62e32b6a234f6dfe6211103e84

**Pull request merged**